### PR TITLE
[bugfix] Fix a bunch of examples to be compatible with dgl 0.5

### PR DESCRIPTION
**Pull request by yzh119**
## Description
As described in the title.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

## Changes
- gat example
- gcn example
- graphsage example
- gcmc example
- [ ]

### CODE REVIEW

File: examples/pytorch/gat/train_ppi.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def main(args):
     g = train_dataset[0]
     n_classes = train_dataset.num_labels
     num_feats = g.ndata['feat'].shape[1]
-    g = g.to(device)
+    g = g.int().to(device)
```
Comment (BarclayII) [inline examples/pytorch/gat/train_ppi.py:63]: We still need to cast it to int32?

Comment (yzh119) [reply examples/pytorch/gat/train_ppi.py:63]: I think so.

Comment (BarclayII) [reply examples/pytorch/gat/train_ppi.py:63]: Hmm how much was the performance gain?  It's OK if you didn't benchmark it though; I'm just curious.  We may need to benchmark it later after 0.5.  I could imagine that newcomers will definitely ask why `g.int()` is necessary and how much the benefit is.

**Comment by VoVAllen** [comment]
Also this https://github.com/dmlc/dgl/issues/1947 @BarclayII

**Comment by BarclayII** [comment]
Should also fix #1947

File: examples/pytorch/gat/train_ppi.py (hunk 1 above)
Comment (yzh119) [reply examples/pytorch/gat/train_ppi.py:63]: For GATConv there is no performance gain, for SAGEConv and GraphConv where cusparse is invoked the performance gap is around 30% and I think the gap would be reduced after we merge tvm kernel.

**Pull request merged**